### PR TITLE
mohammad/onfido_service_token

### PR DESCRIPTION
**Pull request by binary-hamedanchi**

### CODE REVIEW

File: lib/WebService/Async/Onfido.pm [hunk 1]
```diff
@@ -830,6 +835,73 @@ sub report_list {
...
+sub countries_list {
+    my ($self) = @_;
+
+    $self->ua->GET(SUPPORTED_COUNTRIES_URL)->then(sub {
+        try {
+            my ($res) = @_;
+            my $onfido_countries = decode_json_utf8($res->content);
+
+            my $countries_list = {};
+            $countries_list->{uc(country_code2code($_->{alpha3}, 'alpha-3', 'alpha-2'))} = $_->{supported_identity_report} + 0 for @$onfido_countries;
```
Comment (binarytom) [inline lib/WebService/Async/Onfido.pm]: `map` would be slightly neater, although the end result is the same:

```suggestion
            my %countries_list = map { uc(country_code2code($_->{alpha3}, 'alpha-3', 'alpha-2')) => $_->{supported_identity_report} + 0 } @$onfido_countries;
```

File: lib/WebService/Async/Onfido.pm [hunk 2]
```diff
@@ -659,6 +662,8 @@ service account
 
 Returns a L<Future> which will resolve with the result.
 
+=back
```
Comment (binarytom) [inline lib/WebService/Async/Onfido.pm]: Good catch, but it's in the wrong place - needs to be moved up 2 lines!

File: lib/WebService/Async/Onfido.pm [hunk 3]
```diff
@@ -830,6 +835,73 @@ sub report_list {
...
 }
 
+=head2 countries_list
+
+Returns a hashref containing 2-letter country codes as keys and supporting status
+as their value.
+
+=cut
+
+sub countries_list {
```
Comment (binarytom) [inline lib/WebService/Async/Onfido.pm]: Could probably cache this, since the values don't change regularly?

Comment (binary-hamedanchi) [reply lib/WebService/Async/Onfido.pm]: It's being cached in [RPC](https://github.com/regentmarkets/bom-rpc/pull/2035/files#diff-34235887f3ed9d7f84ca919296355eaeR152) and [Events](https://github.com/regentmarkets/bom-events/pull/75/files#diff-0355a08436e02a46826853bc40525f91R892), as we don't have access to Redis here.

File: lib/WebService/Async/Onfido.pm (hunk 1 above)
Comment (binary-hamedanchi) [reply lib/WebService/Async/Onfido.pm]: Great, thanks. 👍

**Pull request merged**